vru_use_case_scripts/plot_detected_objects

```diff
@@ -0,0 +1,251 @@
+)
+
+parser.add_argument(
+    "--vehicle-detection-csv",
+    type=Path,
+    help="CSV file containing object detection entries from a rosbag file",
+    required=True,
+)
+
+parser.add_argument(
+    "--infrastructure-sensors-json",
+    type=Path,
+    help="JSON file containing infrastructure sensor positions",
+    required=True,
+)
+
+cli_args = parser.parse_args()
+
+detections_df = parse_positions_from_detection_csv(cli_args.vehicle_detection_csv)
+sdsm_df = parse_positions_from_sdsm_csv(cli_args.sdsm_csv)
+
+sensor_position = parse_infrastructure_sensor_position_from_json(
+    cli_args.infrastructure_sensors_json
+)
+
+figures_dir = Path("figures")
+figures_dir.mkdir(exist_ok=True)
+
+for detection_id in get_unique_object_ids(detections_df):
+    object_detection = detections_df.loc[detections_df["object_id"] == detection_id]
+    object_sdsm = sdsm_df.loc[sdsm_df["object_id"] == detection_id]
+
+    _, ax = plt.subplots()
+
+    ax.plot(
+        object_detection["map_position_x_m"],
+        object_detection["map_position_y_m"],
+        label="Vehicle Detected",
+    )
+
+    ax.plot(
+        object_sdsm["x"] + sensor_position["x"],
+        object_sdsm["y"] + sensor_position["y"],
+        label="SDSM Reported",
+    )
+
+    ax.set_title(f"Object Trace - {detection_id}")
+    ax.set_ylabel("$y$ (map) [m]")
+    ax.set_xlabel("$x$ (map) [m]")
+    ax.legend()
+
+    ax.set_aspect("equal")
+
+    plt.savefig(figures_dir / f"object_trace_{detection_id}.png")
+
+for detection_id in get_unique_object_ids(detections_df):
+    object_detection = detections_df.loc[detections_df["object_id"] == detection_id]
+    object_sdsm = sdsm_df.loc[sdsm_df["object_id"] == detection_id]
+
+    object_detection_unique_times = object_detection.drop_duplicates("cdasim_time_ms")
+    object_sdsm_unique_times = object_sdsm.drop_duplicates("cdasim_time_ms")
+
+    times = np.arange(object_detection_unique_times["cdasim_time_ms"].iloc[0], object_sdsm_unique_times["cdasim_time_ms"].iloc[-1], 0.01)
+
+    fig, (position_x, error_x, position_y, error_y) = plt.subplots(4, sharex=True)
+
+    position_x.plot(
+        object_detection["cdasim_time_ms"] / 1e3,
+        object_detection["map_position_x_m"],
+        label="Vehicle Detected",
+    )
+
+    position_x.plot(
+        object_sdsm["cdasim_time_ms"] / 1e3,
+        object_sdsm["x"] + sensor_position["x"],
+        label="SDSM Reported"
+    )
+
+    position_x.set_ylabel("$x$ (map) [m]")
+    # position_x.legend()
+    lgd = position_x.legend(bbox_to_anchor=(1.04, 1), borderaxespad=0)
+
+    det_tck_x = splrep(object_detection_unique_times["cdasim_time_ms"].tolist(), object_detection_unique_times["map_position_x_m"].tolist(), k=5)
+    sdsm_tck_x = splrep(object_sdsm_unique_times["cdasim_time_ms"].tolist(), (object_sdsm_unique_times["x"] + sensor_position["x"]), k=5)
+
+    error_x.plot(times / 1e3, abs(BSpline(*det_tck_x)(times) - BSpline(*sdsm_tck_x)(times)))
+    error_x.set_ylabel("Error [m]")
```

Comment:
One of these plots is for y error and the other for x error. Should we include this information in the y axis label or in a plot title or something.